frontend/catalyst/jit.py

```diff
@@ -766,6 +775,28 @@ def closure(qnode, *args, **kwargs):
 
         return jaxpr, out_type, treedef, dynamic_sig
 
+    def get_arg_names(self):
```

Comment:
I'm hesitant to expose another method in the QJIT class as we want to keep it relatively stable and avoid class bloat. Could this be implemented as a free-standing function in the `tracing` submodule (or other jax-related modules if more appropriate, but I think `type_signatures.py` looks pretty relevant)?